docs/guide/writing-tests-for-modules.md

```diff
@@ -0,0 +1,236 @@
+npm install --save-dev apostrophe eslint eslint-config-apostrophe mocha
+```
+
+You will need to add some new scripts to your `package.json`.
+
+```json
+{
+  "name": "article",
+  "version": "1.0.0",
+  "description": "A fictitious Apostrophe module",
+  "main": "index.js",
+  "scripts": {
+    "lint": "npm run eslint",
+    "eslint": "eslint .",
+    "test": "npm run lint && mocha"
+  },
+  "keywords": [],
+  "author": "",
+  "license": "ISC"
+}
+```
+
+Add the following to `.eslintrc.json` to tell eslint to use ApostropheCMS ESLint confiuration.
+
+```json
+{
+  "extends": [ "apostrophe" ]
+}
+```
+
+When you call `npm test`, it will lint your files according to [eslint-config-apostrophe](https://www.npmjs.com/package/eslint-config-apostrophe) rules and run the tests using [mocha](https://www.npmjs.com/package/mocha).
+
+### Test folder
+
+You will need a `test/package.json` file referencing the repository URL of your module. Please replace `article` & `%article-repository-url%` with your module informations.
+
+e.g. for the module [@apostrophecms/login-hcaptcha](https://github.com/apostrophecms/login-hcaptcha) we use 
+
+> `"@apostrophecms/login-hcaptcha": "git://github.com/apostrophecms/login-hcaptcha.git"`
+> 
+
+```json
+{
+  "/**": "This package.json file is not actually installed.",
+  " * ": "Apostrophe requires that all npm modules to be loaded by moog",
+  " */": "exist in package.json at project level, which for a test is here",
+  "dependencies": {
+    "apostrophe": "^3.26.0",
+    "article": "%article-repository-url%"
+  }
+}
+```
+
+You don’t want to commit test files artifacts to git, please add the following to your `.gitignore` file
+
+```gitignore
+# Test files
+/test/apos-build
+/test/public
+/test/locales
+```
+
+## Writing tests
+
+We start with a single test file `test/index.js`.
+
+As your tests grows, you can break it down into multiple files based on your liking.
+
+Mocha does not play well with arrow-functions, more info at [https://mochajs.org/#arrow-functions](https://mochajs.org/#arrow-functions)
+
+<AposCodeBlock>
+  ```javascript
+  const assert = require('assert').strict;
+  const t = require('apostrophe/test-lib/util.js');
+
+  // getAppConfig is used to set options for Apostrophe and the module you want to test
+  const getAppConfig = function (options = {}) {
+    return {
+      article: {
+        options: {
+          // Pass the required options here, if your module doesn't have any options
+          // please skip the `options` attribute
+          ...options,
+          custom: true
+        }
+      }
+    };
+  };
+
+  describe('article', function () {
+    let apos;
+
+    this.timeout(t.timeout);
+
+    before(async function () {
+      apos = await t.create({
+        shortName: 'article',
+        testModule: true,
+        modules: getAppConfig()
+      });
+    });
+
+    after(async function () {
+      await t.destroy(apos);
+    });
+
+    it('should have module options', function () {
```

Comment:
Is each test made up of a description, test function, and assert within the `it`? Can you describe this in the text? Emphasize what other things could be done with the function.